sdks/python/apache_beam/runners/direct/direct_runner.py

```diff
@@ -507,6 +507,17 @@ def run_pipeline(self, pipeline, options):
     from apache_beam.runners.direct.transform_evaluator import \
       TransformEvaluatorRegistry
     from apache_beam.testing.test_stream import TestStream
+    from apache_beam.transforms.external import ExternalTransform
+
+    class VerifyNoCrossLanguageTransforms(PipelineVisitor):
+      """Visitor determining whether a Pipeline uses a TestStream."""
+      def visit_transform(self, applied_ptransform):
+        if isinstance(applied_ptransform.transform, ExternalTransform):
+          raise RuntimeError(
+              "Streaming Python direct runner "
+              "does not support cross-language pipelines.")
```

Comment:
Should there be a FR for this?

Reply:
https://issues.apache.org/jira/browse/BEAM-7514 is probably the closest we have.